core/envelope-sim/src/test/java/fr/sncf/osrd/envelope/ConstraintBuilderTest.java

```diff
@@ -78,6 +77,31 @@ void testSpeedCeilingConstraint() {
         assertEquals(1, builder.lastIntersection);
     }
 
+    @Test
+    void testSpeedMaintainConstraint() {
+        var partBuilder = new EnvelopePartBuilder();
+        var constraint = new SpeedConstraint(2, MAINTAIN_SPEED);
+        var builder = new ConstrainedEnvelopePartBuilder(
+                partBuilder,
+                constraint
+        );
+        assertTrue(constraint.initCheck(1, 2, 2));
+        assertTrue(builder.initEnvelopePart(2, 2, 1));
```

Comment:
Not a comment on the PR itself, you may ignore this

We have two methods that do mostly the same thing, one is called using `(direction, position, speed)`, the other `(position, speed, direction)`. This feels very wrong.

Reply:
I'll just remove this as it was to make codeCov happy, but yeah i'll re-arrange the order of the args